tuned/daemon/controller.py

```diff
@@ -252,6 +254,27 @@ def profile_info(self, profile_name, caller = None):
 			profile_name = self.active_profile()
 		return tuple(self._daemon.profile_loader.profile_locator.get_profile_attrs(profile_name, [consts.PROFILE_ATTR_SUMMARY, consts.PROFILE_ATTR_DESCRIPTION], [""]))
 
+	@exports.export("s", "b")
+	def profile_delete(self, profile, caller = None):
+		if caller == "":
+			return ""
+		if profile == "":
+			print("No set profile to delete.")
+			return False
+		try:
+			profile_list = os.listdir(consts.USER_PROFILE_DIR)
+			if profile not in profile_list:
+				print("File '%s' does not exists." % profile)
+				return False
+			else:
+				shutil.rmtree(os.path.join(consts.USER_PROFILE_DIR, profile))
```

Comment:
I would probably add some escape function sanitizing the final path.

I.e. escape special characters, norm path, check that the abs path is still under USER_PROFILE_DIR and check the the profiledir has tuned.conf, i.e. whether it's valid profile.